newdle/api.py

```diff
@@ -124,6 +125,19 @@ def get_newdle(code):
     return schema_cls().jsonify(newdle)
 
 
+@api.route('/newdle/<code>', methods=('PATCH',))
+@use_args(UpdateNewdleSchema(), locations=('json',))
+def update_newdle(args, code):
+    newdle = Newdle.query.filter_by(code=code).first_or_404('Invalid code')
+    restricted = not g.user or newdle.creator_uid != g.user['uid']
```

Comment:
Since this doesn't have `@allow_anonymous`, you don't need to check for not having `g.user`, so you can just check `if newdle.creator_uid != g.user['uid']: ...` in the next line